validator/keymanager/remote-web3signer/client.go

```diff
@@ -0,0 +1,177 @@
+package remote_web3signer
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/crypto/bls"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+const (
+	ethApiNamespace = "/api/v1/eth2/sign/"
+	maxTimeout      = 5 * time.Second
+)
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type client struct {
+	BasePath   string
+	restClient HTTPClient
+}
+
+func newClient(endpoint string) (*client, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "Invalid Format, unable to parse url")
+	}
+	return &client{
+		BasePath: u.Host,
+		restClient: &http.Client{
+			Timeout: maxTimeout,
+		},
+	}, nil
+}
+
+type SignRequest struct {
+	Type         string        `json:"type"`
+	ForkInfo     *ForkInfo     `json:"fork_info"`
+	SigningRoot  string        `json:"signingRoot"`
+	RandaoReveal *RandaoReveal `json:"randao_reveal"`
+}
+
+type ForkInfo struct {
+	Fork                  *Fork  `json:"fork"`
+	GenesisValidatorsRoot string `json:"genesis_validators_root"`
+}
+
+type Fork struct {
+	PreviousVersion string `json:"previous_version"`
+	CurrentVersion  string `json:"current_version"`
+	Epoch           string `json:"epoch"`
+}
+
+type RandaoReveal struct {
+	Epoch string `json:"epoch"`
+}
+
+type signResponse struct {
+	Signature string `json:"signature"`
+}
+
+func (client *client) Sign(pubKey string, request *SignRequest) (bls.Signature, error) {
+	requestPath := ethApiNamespace + pubKey
+	jsonRequest, err := json.Marshal(request)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid format, failed to marshal json request")
+	}
+	resp, err := client.doRequest(http.MethodPost, client.BasePath+requestPath, bytes.NewBuffer(jsonRequest))
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(resp.Body)
+	jsonDataFromHttp, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to read response body")
+	}
+	signResp := &signResponse{}
+	if err := json.Unmarshal(jsonDataFromHttp, signResp); err != nil {
+		return nil, errors.Wrap(err, "Invalid Format, failed to unmarshal json response")
+	}
```

Comment:
This can probably be its own func as well called unmarshalResponse which takes in something to unmarshal into, such as `target interface{}`